React/my-app/src/screens/book_detail/index.js

```diff
@@ -2,35 +2,78 @@ import React, { Component } from "react";
 import { connect } from "react-redux";
 import defaultImg from "../../assets/photos/bigDefault.png";
 import Book_detail from "./layout.js";
-import { getBook, loadBook, resetBookView } from "../../redux/book/actions";
+import { getBook, resetBookView } from "../../redux/book/actions";
+import { bookAvailable, userHasTheBook } from "../../utils/arrayUtils";
+import {
+  getBookStatus,
+  getWishlist,
+  newIntemWishlist,
+  loading
+} from "../../redux/rents/actions";
+import { NOTHING, NO_SE_ENCUENTRA, DEVOLVER_ANTES } from "./strings";
 
 class BookDetailContainer extends Component {
-  state = {
-    author: null,
-    title: null,
-    genre: null,
-    year: null,
-    image: defaultImg
-  };
+  state = { bookAvailable: false, disabled: false, text: NOTHING };
 
   componentWillMount() {
     this.props.dispatch(getBook(this.props.match.params.id));
+    this.props.dispatch(getBookStatus(this.props.match.params.id));
   }
 
   componentWillUnmount() {
     this.props.dispatch(resetBookView());
   }
 
+  componentWillReceiveProps(newProps) {
+    if (bookAvailable(newProps.bookStatus)) {
+      this.setState({
```

Comment:
Changing state in componentWillReceiveProps is *generally* not necessary.
Here you can compute this "state" in the render based on props. So this component shouldn't need state.

Also react will call `componentWillReceiveProps` even if the props didn't change so you always need to check if `newProps` is different to `this.props` (or at least the props that you are interested in).